docs/WIRELESS.md

```diff
@@ -0,0 +1,162 @@
+# Support for wireless connectivity in EVE
+
+Given that the biggest use case for EVE is in Edge Computing and IoT it is no surprise that wireless connectivity plays a major role in connecting edge applications. Currently EVE supports 2 kinds of wireless networking devices: WiFi cards and Cellular GSM Modems. When it comes to utilizing either of these EVE relies on its controller to supply two critical pieces of information:
+
+* Name of the wireless network to be joined ([SSID](https://en.wikipedia.org/wiki/Service_set_(802.11_network)#SSID) for WiFi and [APN](https://en.wikipedia.org/wiki/Access_Point_Name) for Cellular).
+* Pre-shared secret required to joining the network ([WPA/WPA2/WEP](https://en.wikipedia.org/wiki/Wi-Fi_Protected_Access) and [username/password](https://www.digi.com/resources/documentation/Digidocs/90001399-13/tasks/t-set-up-and-configure-mobile-connectivity.htm) for Cellular).
+
+This dependency on a controller creates a bit of a chicken-and-an-egg problem when the wireless connectivity itself happens to be the only means of reaching the controller. In those cases EVE uses pre-loaded controller configuration to bootstrap the entire process.
```

Comment:
Really glad to see this raised up-front. How does the device know if it should be going down wifi vs cellular vs wired, and if it needs any extra credentials to connect to the network?

Reply:
Makes sense to add a reference to ./CONFIG.md which describes that json file.
But we must also request @naiming-zededa to add a WiFi and WLAN example to CONFIG.md.

Reply:
@rvs did you see this and my other comments about missing references to HARDWARE-BRINGUP and FIRMWARE.md?

Reply:
I've been trying to write a treatise @nordmark ;-) However, you're right an MVP now is better than treatise tomorrow. Please take a look at the MVP of that doc that's available in the updated PR.